src/app/store/slices/ui/index.ts

```diff
@@ -123,6 +125,9 @@ export const uiSlice = createSlice({
     setIsSurveyDialogOpen: (state: UISliceState, action: PayloadAction<boolean>) => {
       state.isSurveyDialogOpen = action.payload;
     },
+    setIsPreferencesDialogOpen: (state: UISliceState, action: PayloadAction<boolean>) => {
+      state.isPreferencesDialogOpen = action.payload;
+    },
```

Comment:
If it is not necessary don't add the state to open the modal to redux, if it is only going to be managed from App.tsx you can use a local state. Using redux for that is overkill.

Reply:
Needed for other components